framework/src/main/java/org/checkerframework/common/purity/PurityChecker.java

```diff
@@ -0,0 +1,24 @@
+package org.checkerframework.common.purity;
+
+import java.util.LinkedHashSet;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+
+/**
+ * Perform purity checking only.
+ *
+ * @checker_framework.manual #type-refinement-purity Side effects, determinism, purity, and
+ *     flow-sensitive analysis
+ */
+public class PurityChecker extends BaseTypeChecker {
+
+    @Override
+    protected BaseTypeVisitor<?> createSourceVisitor() {
+        return new PurityVisitor(this);
+    }
+
+    @Override
+    protected LinkedHashSet<Class<? extends BaseTypeChecker>> getImmediateSubcheckerClasses() {
```

Comment:
A comment here about why this override is necessary (you don't want to always add the purity checker as a subchecker!) would be helpful.

Reply:
It would also be good to test the interaction between the methodval checker and this new purity checker to make sure that the old purity logic wasn't relying on the methodval checker to resolve reflection in any cases. As written, the purity checker can never run the methodval checker as a subchecker, even if `shouldResolveReflection()` would return true.

Or maybe this override should still contain that logic? That worries me, because it suggests that the similar override in the methodval checker should contain a check for purity.